core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -163,6 +163,8 @@ class TaskManager(taskManagerResources: TaskManagerResources = TaskManagerDefaul
   override protected def scriptPathFor(task: Task, taskId: TaskId): Path = pathFor(task, taskId, actualScriptsDirectory, "sh")
   override protected def logPathFor(task: Task, taskId: TaskId): Path = pathFor(task, taskId, actualLogsDirectory, "log")
 
+  def joinOnRunningTasks(millis: Long) = taskExecutionRunner.joinAll(millis)
```

Comment:
Either write some doc and consider putting this method into one of the `TaskManager` traits (probably don't), or better yet, make it `private[execsystem]` and indicate it's for testing.